File: 2. Godina/CS130 - C-C++ programski jezik/Domaci/DZ08/Student.cpp
#include "Student.h"
#include <iostream>

using namespace std;

Student::Student() : ime(""), prezime(""), prosek(0.0) {}

Student::Student(const string& ime, const string& prezime, float prosek)
    : ime(ime), prezime(prezime), prosek(prosek) {}

Student::Student(const Student& other) : ime(other.ime), prezime(other.prezime), prosek(other.prosek) {}

Student::~Student() {
    cout << "Objekat je unisten" << endl;
}

void Student::setIme(const string& ime) {
    this->ime = ime;
}

void Student::setPrezime(const string& prezime) {
    this->prezime = prezime;
}

void Student::setProsek(float prosek) {
    this->prosek = prosek;
}

string Student::getIme() const {
    return ime;
}

string Student::getPrezime() const {
    return prezime;
}

float Student::getProsek() const {
    return prosek;
}

File: 2. Godina/CS203 - Algoritmi i strukture podataka/Ispit/Svi Zadaci - Resenja/Resenja/src/main/java/fajlovi/zadatak8/Student.java
package fajlovi.zadatak8;

public class Student {

    private String indeks;
    private String ime;
    private String prezime;
    private int prosek;

    public Student() {

    }

    public Student(String indeks, String ime, String prezime, int prosek) {
        this.indeks = indeks;
        this.ime = ime;
        this.prezime = prezime;
        this.prosek = prosek;
    }

    public String getIndeks() {
        return indeks;
    }

    public void setIndeks(String indeks) {
        this.indeks = indeks;
    }

    public String getIme() {
        return ime;
    }

    public void setIme(String ime) {
        this.ime = ime;
    }

    public String getPrezime() {
        return prezime;
    }

    public void setPrezime(String prezime) {
        this.prezime = prezime;
    }

    public int getProsek() {
        return prosek;
    }

    public void setProsek(int prosek) {
        this.prosek = prosek;
    }

    @Override
    public String toString() {
        return "Student{" + "indeks=" + indeks + ", ime=" + ime + ", prezime=" + prezime + ", prosek=" + prosek + '}';
    }

}
